src/dds/ReaderWriter_dds.cpp

```diff
@@ -0,0 +1,285 @@
+                layout.format = it->second;
+                layout.maxNumMipmaps = numMipMaps;
+
+                switch (dim)
+                {
+                case tinyddsloader::DDSFile::TextureDimension::Texture1D:
+                    vsg_data = vsg::ubvec4Array::create(width, reinterpret_cast<vsg::ubvec4*>(raw), layout);
+                    break;
+                case tinyddsloader::DDSFile::TextureDimension::Texture2D:
+                    if (isCubemap && numArrays == 6)
+                        vsg_data = vsg::ubvec4ArrayCube::create(width, height, reinterpret_cast<vsg::ubvec4*>(raw), layout);
+                    else
+                        vsg_data = vsg::ubvec4Array2D::create(width, height, reinterpret_cast<vsg::ubvec4*>(raw), layout);
+                    break;
+                case tinyddsloader::DDSFile::TextureDimension::Texture3D:
+                    vsg_data = vsg::ubvec4Array3D::create(width, height, depth, reinterpret_cast<vsg::ubvec4*>(raw), layout);
+                    break;
+                }
+
+                //std::cout << "* Finish: " << valueCount * valueSize << ", " << vsg_data->dataSize() << std::endl;
+
+                return vsg_data;
+            }
+        }
+        else
+        {
+            std::cerr << "Format is not supported yet: " << (uint32_t)format << std::endl;
+        }
+
+        return {};
+    }
+}
+
+using namespace vsgXchange;
+
+ReaderWriter_dds::ReaderWriter_dds() :
+    _supportedExtensions{"dds"}
+{
+}
+
+vsg::ref_ptr<vsg::Object> ReaderWriter_dds::read(const vsg::Path& filename, vsg::ref_ptr<const vsg::Options> /*options*/) const
+{
+    if (!vsg::fileExists(filename))
+        return {};
+
+    if (const auto ext = vsg::fileExtension(filename); _supportedExtensions.count(ext) == 0)
+        return {};
+
+    tinyddsloader::DDSFile ddsFile;
+
+    if (const auto result = ddsFile.Load(filename.c_str()); result == tinyddsloader::Success)
+    {
+        return readDds(ddsFile);
+    }
+    else
+    {
+        switch (result)
+        {
+        case tinyddsloader::ErrorNotSupported:
+            std::cerr << "Error loading file: Feature not supported" << std::endl;
+            break;
+        default:
+            std::cerr << "Error loading file: " << result << std::endl;
+            break;
+        }
+    }
+
+    return {};
+}
+
+vsg::ref_ptr<vsg::Object> ReaderWriter_dds::read(std::istream& fin, vsg::ref_ptr<const vsg::Options> options) const
+{
+    if (_supportedExtensions.count(options->extensionHint) == 0)
+        return {};
+
+    std::vector<uint8_t> buffer(1 << 16, 0); // 64kB
+    std::vector<uint8_t> input;
+
+    while (!fin.eof())
+    {
+        fin.read((char*)&buffer[0], buffer.size());
+        const auto bytes_readed = fin.gcount();
+        input.insert(input.end(), buffer.begin(), buffer.end());
+    }
+
+    tinyddsloader::DDSFile ddsFile;
+
+    if (const auto result = ddsFile.Load(std::move(input)); result == tinyddsloader::Success)
```

Comment:
@anormann1974 I'm currently doing a review of all the changes between master and KTX and have been reflecting on the approach of copying an istream to a std::string then passing on as memory to the loader, in this case to ddsFile.Load(std::move(input)).  There is a Load(istream) as well so I think the in this case it would be more natural to use that.